internal/memoize/cache.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2023 Juan Pablo Tosso and the OWASP Coraza contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build !tinygo
+
+// Highly inspired in https://github.com/patrickmn/go-cache/blob/master/cache.go
```

Comment:
This is ok but this file looks like an obvious mutex guarded map, not sure it's worth referencing anything.

Anyways it looks like a `sync.Map` should be much faster for this, it's docs specifically mention the write once read many case of a cache